src/datepicker/datepicker.js

```diff
@@ -509,6 +509,7 @@ function ($compile, $parse, $document, $position, dateFilter, dateParser, datepi
           ngModel.$setValidity('date', true);
           return viewValue;
         } else if (angular.isString(viewValue)) {
+          if (!dateFormat) return null;
```

Comment:
Should initialize var dateFormat = datepickerPopupConfig.datepickerPopup instead.